### PR TITLE
drivers/sensors: add support of InvenSense MPU-9250 sensor

### PR DESCRIPTION
**Pull request by zouboan**
## Summary
Implement drivers  of MPU-9250 in new sensor framework, include Accelerometer, Gyroscope, Magnetometer
## Impact
nothing
## Testing
Custom board and mpu9250 on SPI bus

sensortest accel0
sensortest gyro0
sensortest mag0

uorb_listener

### CODE REVIEW

File: drivers/sensors/mpu9250.c [hunk 1]
```diff
@@ -0,0 +1,2045 @@
...
+        break;
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: mpu9250_accel_scale
+ *
+ * Description: Set scale of accleratormeter.
```
Comment (acassis) [inline drivers/sensors/mpu9250.c:507]:
```suggestion
 * Description: Set scale of accelerometer.

File: drivers/sensors/mpu9250.c [hunk 2]
```diff
@@ -0,0 +1,2045 @@
...
+        break;
+      default:
+        break;
+    }
+}
+
+/****************************************************************************
+ * Name: mpu9250_gyro_scale
+ *
+ * Description: Set scale of accleratormeter.
```
Comment (acassis) [inline drivers/sensors/mpu9250.c:545]:
```suggestion
 * Description: Set scale of accelerometer.

**Comment by acassis** [comment]
Very nice @zouboan! I suggest you to include later a board usage example for some known, low cost and easy to find board like stm32f4discovery, raspberrypi-pico or esp32-devkit.

**Comment by zouboan** [comment]
@acassis Thanks, i will explorer which board have MPU9250

File: drivers/sensors/mpu9250.c [hunk 3]
```diff
@@ -0,0 +1,2045 @@
...
+              /* Set gyroscope full scale */
+
+              case SENSOR_TYPE_GYROSCOPE:
+                {
+                  ret = mpu9250_write_gyro_range(dev, (uint8_t)arg);
+                  mpu9250_gyro_scale(priv, (enum gyro_config_bit)arg);
+                }
+                break;
+
+              /* Set accleratormeter full scale */
```
[the remark this answers is not in the document]
Comment (zouboan) [reply drivers/sensors/mpu9250.c]: done

File: drivers/sensors/mpu9250.c (hunk 1 above)
Comment (zouboan) [reply drivers/sensors/mpu9250.c]: done

File: drivers/sensors/mpu9250.c (hunk 2 above)
Comment (zouboan) [reply drivers/sensors/mpu9250.c]: done

**Pull request merged**